token/program-2022/src/state.rs

```diff
@@ -289,6 +290,18 @@ fn unpack_coption_u64(src: &[u8; 12]) -> Result<COption<u64>, ProgramError> {
     }
 }
 
+// `spl_token_program_2022::extension::AccountType::Account` ordinal value
+const ACCOUNTTYPE_ACCOUNT: u8 = AccountType::Account as u8;
+impl GenericTokenAccount for Account {
+    fn valid_account_data(account_data: &[u8]) -> bool {
+        spl_token::state::Account::valid_account_data(account_data)
+            || ACCOUNTTYPE_ACCOUNT
+                == *account_data
+                    .get(spl_token::state::Account::get_packed_len())
+                    .unwrap_or(&0)
+    }
+}
+
```

Comment:
Can you please add test cases for `unpack_account_mint()` and `unpack_account_owner` for an extended token account?

Reply:
Will do.

Reply:
Done